client/src/pages/StudyGroupsPage.js

```diff
@@ -1,3 +1,146 @@
-const StudyGroupPage = () => {};
+import SideBar from "../components/SideBar";
+import {
+  Button,
+  Card,
+  CardBody,
+  CardFooter,
+  Dialog,
+  Input,
+  Typography,
+} from "@material-tailwind/react";
+import {
+  ArrowLongRightIcon,
+  ChatBubbleLeftRightIcon,
+} from "@heroicons/react/24/solid";
+import { v4 as uuidv4 } from "uuid";
+import { useDispatch, useSelector } from "react-redux";
+import { postChatHistoryAsync } from "../store/chat/thunks";
+import { getUserAsync, groupChatAsync } from "../store/authentication/thunks";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import ChatMembers from "../components/chat/chatMembers";
+
+const StudyGroupPage = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const user = useSelector((state) => state.loginReducer.user);
+  const [chatName, setChatName] = useState("");
+  const [openCreateChat, setOpenCreateChat] = useState(false);
+
+  useEffect(() => {
+    if (user) {
+      dispatch(getUserAsync(user.userID));
+    }
+  }, [dispatch]);
+
+  const createChatHandler = () => {
+    setOpenCreateChat(!openCreateChat);
+  };
+
+  const onClickHandler = (event) => {
+    if (chatName.length < 1) return;
+
+    event.preventDefault();
+    setOpenCreateChat(!openCreateChat);
+    const id = uuidv4();
+
+    const initChat = {
+      groupID: id,
+      name: chatName,
+    };
+
+    const updatedUser = {
+      username: user.username,
+      groupID: id,
+    };
+
+    dispatch(groupChatAsync(updatedUser));
+    dispatch(postChatHistoryAsync(initChat));
+    navigate(`/chat/${id}`);
+  };
+
+  if (!user) {
+    return null;
+  }
+
+  return (
+    <div>
+      <div className="fixed left-0 top-0 z-[1035] h-screen">
+        <SideBar />
+      </div>
+      <div className="p-5 !pl-[300px]">
+        <div className="mx-5">
+          <Button onClick={createChatHandler} className="bg-indigo-300 m-2">
+            Create Chat
+          </Button>
+          <div className="flex flex-wrap justify-center">
+            {user && user.groupID
+              ? user.groupID.map((group, index) => (
+                  <Card className="mt-6 w-80 m-5" key={index}>
+                    <CardBody className="pb-2">
+                      <ChatBubbleLeftRightIcon className="text-indigo-300 w-12 h-12 mb-4" />
+                      <Typography
+                        variant="h5"
+                        color="blue-gray"
+                        className="mb-2"
+                      >
+                        Group Chat: {group}
+                      </Typography>
+                      <ChatMembers groupID={group} />
```

Comment:
This reveals user first and last name to the chat group which some users may not be comfortable with. Right now there is also no chance for the user to decline an invite as well.

Reply:
Do you suggest displaying usernames instead? 
can definitely add decline/accept invite functionality in the future

Reply:
Yeah: the invite and chat messages use username as well so the inviter wouldn't be given any new information about the people they invite.

Reply:
done